### PR TITLE
Adding 'n/a' to list of strings denoting missing values

### LINKED ISSUE #16078: 'n/a' is not among default list of missing value strings
`n/a` denotes missing values in the [Brain Imaging Data Structure (BIDS)](https://www.nature.com/articles/sdata201644) standard. Adding it to the list of strings denoting missing values would make using pandas on BIDS datasets easier.

### PR DESCRIPTION
**Pull request by chrisgorgo**
- [x] closes #16078
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

**Comment by chrisfilo** [comment]
Wow. That codecov report makes little sense.

**Comment by jreback** [comment]
I suppose this is ok.

@jorisvandenbossche

File: doc/source/whatsnew/v0.20.0.txt [hunk 1]
```diff
@@ -521,6 +521,7 @@ Other Enhancements
 - The ``usecols`` argument in ``pd.read_csv()`` now accepts a callable function as a value  (:issue:`14154`)
 - The ``skiprows`` argument in ``pd.read_csv()`` now accepts a callable function as a value  (:issue:`10882`)
 - The ``nrows`` and ``chunksize`` arguments in ``pd.read_csv()`` are supported if both are passed (:issue:`6774`, :issue:`15755`)
+- ``pd.read_csv()`` now treats 'n/a' strings as missing values by default (:issue:`16078`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.20.0.txt]: use ``n/a``

Comment (chrisfilo) [inline doc/source/whatsnew/v0.20.0.txt]: Thanks for the feedback, but could you clarify what exactly would you like me to change?

Comment (TomAugspurger) [inline doc/source/whatsnew/v0.20.0.txt]: Put double backticks around it so it's formatted as code in the HTML (but keep the single-quotes too I think)

**Comment by chrisfilo** [comment]
Done.

**Comment by jreback** [comment]
@jorisvandenbossche @TomAugspurger ?

lgtm.

**Comment by TomAugspurger** [comment]
I think +1... My only hesitation is that we've run into problems before with people un-wantingly getting stuff converted to NaN, but since we already treat N/A as a nan value, I think this is fine. @jorisvandenbossche thoughts?

**Comment by jorisvandenbossche** [comment]
Yeah, if we would start again, certainly +1 on adding it to the default list to parse as missing. But if someone has csv files with actual valid 'n/a' values, we will for sure break his/her code ... Difficult to assess however if that would occur. 
I will say, if you are +1, I won't object :-)

**Comment by jorisvandenbossche** [comment]
Small comment on the PR: I would list this in the backwards incompatible API changes section, not enhancements

**Comment by TomAugspurger** [comment]
Since this is technically an API change, I think we should wait till 0.20.1. Sorry for not getting this in before the RC @chrisfilo

**Comment by jorisvandenbossche** [comment]
> Since this is technically an API change, I think we should wait till 0.20.1.

Small correction, I think this should be 0.21.0, as 0.20.1 shouldn't contain API changes.

**Comment by chrisfilo** [comment]
So how should we proceed? Is there a separate branch for 0.21.0 I should send thi PR to?

**Comment by TomAugspurger** [comment]
> So how should we proceed? Is there a separate branch for 0.21.0 I should send thi PR to?

It will still be merged into master, just not until after we've finished the 0.20.0 release (this week sometime). Once that's out, I'll add a `whatsnew/0.21.0` and ask you to move your release note over there.

**Comment by gfyoung** [comment]
Also, need more tests!  Try passing in data containing your new `na_value`.

**Comment by jreback** [comment]
an you rebase

**Comment by chrisfilo** [comment]
Done. @gfyoung the test you requested is here: https://github.com/pandas-dev/pandas/pull/16079/files#diff-6e435422c67fa1384140f92110fb69a7R73

**Comment by gfyoung** [comment]
Awesome!  @jreback @TomAugspurger might it be prudent to start sharing the na_values list somehow so that we don't have to do maintenance in two different files?  Not blocking this PR but jut a thought.

**Comment by jreback** [comment]
> Awesome! @jreback @TomAugspurger might it be prudent to start sharing the na_values list somehow so that we don't have to do maintenance in two different files? Not blocking this PR but jut a thought.

yeah that is certainly possible, if you'd like to do an issue/PR

**Comment by chrisfilo** [comment]
I'm getting the following test error, but I don't see how it could be related to the changes I made. Please advise:

```
=================================== FAILURES ===================================
_ TestCommonIOCapabilities.test_write_fspath_all[to_hdf-writer_kwargs3-tables] _

self = <pandas.tests.io.test_common.TestCommonIOCapabilities object at 0x7ff8c30fdc88>
writer_name = 'to_hdf', writer_kwargs = {'key': 'bar', 'mode': 'w'}
module = 'tables'

    @pytest.mark.parametrize('writer_name, writer_kwargs, module', [
        ('to_csv', {}, 'os'),
        ('to_excel', {'engine': 'xlwt'}, 'xlwt'),
        ('to_feather', {}, 'feather'),
        ('to_hdf', {'key': 'bar', 'mode': 'w'}, 'tables'),
        ('to_html', {}, 'os'),
        ('to_json', {}, 'os'),
        ('to_latex', {}, 'os'),
        ('to_msgpack', {}, 'os'),
        ('to_pickle', {}, 'os'),
        ('to_stata', {}, 'os'),
    ])
    def test_write_fspath_all(self, writer_name, writer_kwargs, module):
        p1 = tm.ensure_clean('string')
        p2 = tm.ensure_clean('fspath')
        df = pd.DataFrame({"A": [1, 2]})
    
        with p1 as string, p2 as fspath:
            pytest.importorskip(module)
            mypath = CustomFSPath(fspath)
            writer = getattr(df, writer_name)
    
            writer(string, **writer_kwargs)
            with open(string, 'rb') as f:
                expected = f.read()
    
            writer(mypath, **writer_kwargs)
            with open(fspath, 'rb') as f:
                result = f.read()
    
>           assert result == expected
E           AssertionError: assert b'\x89HDF\r\n...0\x00\x00\x00' == b'\x89HDF\r\n\...0\x00\x00\x00'
E             At index 2620 diff: 164 != 163
E             Use -v to get the full diff
```

**Comment by gfyoung** [comment]
Do you get this failure consistently?  And with which parameters is it failing?  (I presume `to_csv` but want to double check)

**Comment by chrisfilo** [comment]
Actually it's not `to_csv`, but `to_hdf`. If you could trigger rebuild on Circle we can check if this is failing consistently (sorry I don't have permissions to do it): https://circleci.com/gh/pandas-dev/pandas/2031#tests/containers/2

**Comment by gfyoung** [comment]
@chrisfilo : Neither do I (@jreback ?) :sad: However, I have been noticing lately that `CircleCI` has been acting up on us and failing in random places like this.  I'm beginning to wonder if `CircleCI` is useful for our testing purposes compared to Travis and Appveyor.

**Comment by chrisfilo** [comment]
ping

**Comment by jreback** [comment]
can you rebase this

File: doc/source/whatsnew/v0.21.0.txt [hunk 2]
```diff
@@ -49,6 +49,7 @@ Backwards incompatible API changes
 
 - Accessing a non-existent attribute on a closed :class:`HDFStore` will now
   raise an ``AttributeError`` rather than a ``ClosedFileError`` (:issue:`16301`)
+- ``pd.read_csv()`` now treats ``'n/a'`` strings as missing values by default (:issue:`16078`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: can you move this right below the other issue (where null is added)

**Comment by chrisfilo** [comment]
Here you go.

PS Hint for keeping contributors happy - merge conflicting PRs in chronological order so people will not be punished with resolving conflicts for no apparent reason.

File: doc/source/whatsnew/v0.21.0.txt (hunk 2 above)
Comment (chrisfilo) [inline doc/source/whatsnew/v0.21.0.txt]: I could, but @jorisvandenbossche made exactly the opposite request: https://github.com/pandas-dev/pandas/pull/16079#issuecomment-297853157

**Comment by jreback** [comment]
> PS Hint for keeping contributors happy - merge conflicting PRs in chronological order so people will not be punished with resolving conflicts for no apparent reason.

hah!

yeah we do leave space in the whatsnew so that we won't have conflicts to begin with.

we have lots of PR's. It is an iterative process. I am a bit biased as I look for recently updated ones that are green. Then I (or others) may make comments, causing another round of changes. The more responsive people are, the quicker things get merged. And of course the more they rebase the less conflicts occur.

:.

lgtm. ping on green.

**Comment by jreback** [comment]
> PS Hint for keeping contributors happy - merge conflicting PRs in chronological order so people will not be punished with resolving conflicts for no apparent reason.

this actually is not fair. more fair is merging responsive PR's. They are more likely to be ready. We have long running PR's which need rebasing, which pro-active authors do. 

Sure when we have a queue and/or are trying to get out a release things get a bit jumbled...

**Comment by chrisfilo** [comment]
I believe in this case the more considerate thing to do would've been to read the [comments](https://github.com/pandas-dev/pandas/pull/16079#issuecomment-302888648), hit the rebuild button on circle, wait for green and merge this first. Recency of the last comment is a poor proxy of responsiveness if the contributor is blocked on faulty CI. This incredibly simple PR was ready to merge 12 days ago.

Ultimately it's is the faulty CI (or over reliance on it) that is causing the frustrations. I understand that there is a shit tonne of PRs in this project (truly remarkable), but I'm trying to give you a unique perspective of a project newcomer and an ability to improve your practices. Overall goal is to make the community better and more efficient.

**Comment by jreback** [comment]
> I believe in this case the more considerate thing to do would've been to read the comments, hit the rebuild button on circle, wait for green and merge this first. Recency of the last comment is a poor proxy of responsiveness if the contributor is blocked on faulty CI. This incredibly simple PR was ready to merge 12 days ago.

@chrisfilo well you are assuming we are contantly searching for PR's that are ready to merge :> this is not the case. rather we DO get a ping and then react, or I do scan things on ready to go PR's. I DO prioritize all green PR's as those are most likely to be ready. Sure a CI failure happens, and certainly if pinged we would restart if possible, but if it didn't get restarted then its a red X, and it appears like the PR is non-responsive. Of could the writer can always do a ``git commit -C HEAD --all --amend`` and push to force a rebuild :>

**Comment by jreback** [comment]
> pandas/tests/io/parser/na_values.py:73:80: E501 line too long (83 > 79 characters)

flake issue

**Comment by chrisfilo** [comment]
You were [pinged](https://github.com/pandas-dev/pandas/pull/16079#issuecomment-302892117).
[Twice](https://github.com/pandas-dev/pandas/pull/16079#issuecomment-305571589).

But you are right that I could've just done `git commit -C HEAD --all --amend`.

File: doc/source/whatsnew/v0.21.0.txt (hunk 2 above)
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: ahh I see. then I will move the other one here :>

**Comment by jreback** [comment]
thanks!

**Pull request merged**